src/server/rpc/procedures/roboscape/firmware/roboscape.c

```diff
@@ -33,6 +37,17 @@ static const unsigned char server_port[2] = { 0x07, 0xb5 }; // 1973
 
 unsigned int time_ref = 0;
 unsigned int last_cnt = 0;
+int comSeqNum = 0;
+
+
+// to keep track of time for user button
+unsigned int buttonState;     // current state of the button
+unsigned int lastButtonState = 1; // previous state of the button
+unsigned int startPressed = 0;    // the time button was pressed
+unsigned int endPressed = 0;      // the time button was released
+unsigned int timeHold = 0;        // the time button is hold
```

Comment:
correct

Reply:
An enum is always a compile time constant (even with old gcc), the unsigned int will have a runtime representation (especially since it is not const and not static) unless the gcc is doing whole program analysis.